odrive-motor/src/odriveCAN.py

```diff
@@ -0,0 +1,290 @@
+                set_configs(odriveCAN.odrv, odriveCAN.odrive_config_file)
+                rsetattr(odriveCAN.odrv, "axis0.config.can.node_id", odriveCAN.nodeID)
+                odriveCAN.torque_constant = find_motor_configs(odriveCAN.odrive_config_file, "torque_constant")
+                odriveCAN.current_soft_max = find_motor_configs(odriveCAN.odrive_config_file, "current_soft_max")
+        except TimeoutError:
+            LOGGER.warn("Could not set odrive configurations because no serial odrive connection was found.")
+
+        if config.attributes.fields["canbus_baud_rate"].string_value != "":
+            baud_rate = config.attributes.fields["canbus_baud_rate"].string_value
+            odriveCAN.baud_rate = baud_rate[0:len(baud_rate)-1] + "000"
+        elif odriveCAN.odrive_config_file != "":
+            baud_rate = find_baudrate(odriveCAN.odrive_config_file)
+            odriveCAN.baud_rate = str(baud_rate)
+        else:
+            odriveCAN.baud_rate = "250000"
+
+        os.system("sudo ip link set can0 down")
+        os.system("sudo ip link set can0 up type can bitrate " + odriveCAN.baud_rate)
+        # send an arbitrary message because the first message sent over CAN does not go through
+        msg = db.get_message_by_name('Clear_Errors')
+        msg = can.Message(arbitration_id=msg.frame_id | odriveCAN.nodeID << 5, is_extended_id=False)
+        try:
+            bus.send(msg)
+        except can.CanError:
+            pass
+
+        def periodically_surface_errors(odriveCAN):
+            while True:
+                asyncio.run(odriveCAN.surface_errors())
+                time.sleep(1)
+
+        thread = Thread(target = periodically_surface_errors, args=[odriveCAN])
+        thread.setDaemon(True) 
+        thread.start()
+
+        return odriveCAN
+
+    def reconfigure(self, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]):
+        baud_rate = "250000"
+        if config.attributes.fields["canbus_baud_rate"].string_value != "":
+            baud_rate = config.attributes.fields["canbus_baud_rate"].string_value
+            baud_rate = baud_rate[0:len(baud_rate)-1] + "000"
+
+        if baud_rate != self.baud_rate:
+            self.baud_rate = baud_rate
+            os.system("sudo ip link set can0 down")
+            os.system("sudo ip link set can0 up type can bitrate " + self.baud_rate)
+            msg = db.get_message_by_name('Clear_Errors')
+            msg = can.Message(arbitration_id=msg.frame_id | self.nodeID << 5, is_extended_id=False)
+            try:
+                bus.send(msg)
+            except can.CanError:
+                pass
+        
+        new_nodeID = config.attributes.fields["canbus_node_id"].number_value
+        if new_nodeID != self.nodeID:
+            self.set_node_id(new_nodeID)
+
+    async def set_power(self, power: float, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        torque = power*self.current_soft_max*self.torque_constant
```

Comment:
Currently running into an issue where the tuned values for `current_soft_max` and `torque_constant` from the config make the motor only operable at below approx 20% power. Anything higher results in a torque that causes `ODriveError.DC_BUS_UNDER_VOLTAGE`. Also, anything above approx 12% is too fast for the position of the encoder to update. I'm not really sure what an appropriate way to scale the torque would be that would be applicable for all motors. Some things that have worked decently in my testing are dividing the torque input by 100, or using only a portion (such as square root) of the `current_soft_max` when calculating the torque

Reply:
the encoder issues could likely be fixed by increasing the `encoder_msg_rate_ms` value in the config file, which is currently 100 ms, but not sure how high that would need to be to fix the issue. I'll look into it

Reply:
potential solution, use `axis0.config.motor.calibration_current` instead of `axis0.config.motor.current_soft_max`. In the config I'm using, this value is `6.0`, which works well, but I don't know how this would carry over to other tuned motors, or what `calibration_current` 100% means

Reply:
is there a config.current_limit set somewhere? And is the config.calibration_current smaller than that limit? Muy current (heh) thought is that there is something limiting the current draw and so you're not seeing expected behaviour from the torque values we're feeding into set_torque. 

https://docs.odriverobotics.com/v/0.5.4/getting-started.html#setting-the-limits

I wonder if the soft max is actually undercut by another parameter set in calibration then. Maybe since they're defaults for the ODrive, they don't mean much to motors, especially smaller ones.